src/gemdat/rotations.py

```diff
@@ -247,22 +246,49 @@ def set_symmetry_operations(
                 element.rotation_matrix for element in g.symmetry_ops
             ]).transpose(1, 2, 0)
         elif explicit_sym is not None:
+            if explicit_sym.ndim != 3:
+                # reshape if a single transformation is provided
+                explicit_sym = explicit_sym.reshape(1, 3, 3)
             self.sym_matrix = explicit_sym
         else:
             raise ValueError(
                 'At least one of sym_group or explicit_sym must be provided')
 
-    def get_symmetric_trajectory(self, ) -> np.ndarray:
-        """Returns the symmetric trajectory.
+    def set_transformation(self, transformations: list[str] | None = None):
+        """Set the list of transformations to be executed.
+
+        If None, return the original trajectory.
+        """
+        if transformations is None:
+            self.transformations = []
+            self.transformed_trajectory = self.unit_vec_trajectory
+        else:
+            self.transformations = [
+                globals()[name]() for name in transformations
+            ]
+            # Set to None to force re-execution of transformations
+            self.transformed_trajectory = None
+
+    def transform(self,
+                  transformations: list[str] | None = None) -> np.ndarray:
+        """Execute all transformations in the list of transformations.
 
         Returns
         -------
-        direction_sym: np.ndarray
-            Trajectory of the unit vectors after applying symmetry operations
+        transformed_trajectory : np.ndarray
+            the unit vector trajectory after the transformations.
         """
-        if not hasattr(self, '_symmetric_trajectory'):
-            self._symmetric_trajectory = self._compute_symmetric_trajectory()
-        return self._symmetric_trajectory
+        if transformations is None:
+            self.transformed_trajectory = self.unit_vec_trajectory
+        else:
+            t_operators = [globals()[name]() for name in transformations]
+
+            # Restart from the original unit vector trajectory
+            self.transformed_trajectory = self.unit_vec_trajectory
+            # then apply all transformations
+            for transformation in t_operators:
+                transformation.transform(self)
+        return self.transformed_trajectory

```

Comment:
Do you think we could return a new `Orientations` class from `transformation.transform`? I think that will fit very nicely with your strategy pattern.

I think it's also better to use a lookup table for the transformation functions to avoid using `globals()` here in this way. That will give some odd error messages in case users mistype the function name.

The lookup table can look something like this:

```
transform_lookup = {
    'normalize': Normalize.transform,
    'conventional': Conventional.transform,
    'symmetrize': Symmetrize.transform,
}
```

The lookup table also shows why we do not necessarily need classes here (see comment above).

```suggestion
        t_operators = [transform_lookup[name] for name in transformations]

        ret = self
        for t_operator in t_operators:
            ret = t_operator(ret)
        
        return ret
```